File: 2021/0066_plusOne/0811/plusOne.rs
impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let mut tmp = digits;
        let mut i = tmp.len() as i32 - 1;
        loop {
            if i < 0 { break }
            let j = i as usize; // index must be usize type
            tmp[j] += 1;
            tmp[j] %= 10;
            if tmp[j] != 0 {
                return tmp
            }
            i -= 1
        }
        tmp.insert(0, 1);
        tmp
    }
}

File: 2021/0066_plusOne/0812/plusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        for i in range(len(digits) - 1, -1, -1):
            digits[i] = (digits[i] + 1) % 10
            if digits[i] != 0: return digits
        return [1] + digits
